src/main/java/forestry/core/patchouli/processor/CarpenterProcessor.java

```diff
@@ -1,58 +1,34 @@
 package forestry.core.patchouli.processor;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import net.minecraft.world.item.ItemStack;
-import net.minecraft.world.item.crafting.Ingredient;
-
+import com.google.common.base.Preconditions;
 import forestry.api.recipes.ICarpenterRecipe;
 import forestry.api.recipes.RecipeManagers;
-
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
 import vazkii.patchouli.api.IComponentProcessor;
 import vazkii.patchouli.api.IVariable;
 import vazkii.patchouli.api.IVariableProvider;
 
+import javax.annotation.Nullable;
+
+@SuppressWarnings("unused")
 public class CarpenterProcessor implements IComponentProcessor {
+	@Nullable
 	protected ICarpenterRecipe recipe;
 
 	@Override
 	public void setup(IVariableProvider variables) {
-		int index;
-		try {
-			index = variables.get("index").asNumber().intValue();
-		} catch (Exception e) {
-			index = 0;
-		}
-
-		ItemStack itemStack;
-		try {
-			itemStack = variables.get("item").as(ItemStack.class);
-		} catch (Exception e) {
-			itemStack = ItemStack.EMPTY;
-		}
-
-        /*
-        Manually iterate over all the carpenter recipes by result item ResourceLocation id
-        because looking up the recipe in carpenterManager doesn't work for some recipes
-        (idk why, maybe i was doing it wrong)
```

Comment:
@ramidzkh you weren't doing anything wrong here, the existing method just had a really strict equality check for ItemStack. I removed that existing method, because I think ItemStack comparison should always be left up to the caller, since every situation is different.